chesster.js

```diff
@@ -617,446 +256,128 @@ controller.hears([
-                callback("failed to get player: " + team_member.name);
-            }
-        });
-    };
-}
 
-controller.hears([
-	'team members'
-],[
-	'direct_mention', 
-	'direct_message'
-], function(bot, message) {
-    bot_exception_handler(bot, message, function(){
-        var self = this;
-        self.team_name = message.text.split(" ").slice(2).join(" ");
-        if(self.team_name && self.team_name != ""){
-	    loadSheet(self, function(){
-	        getMembers(self, function(){
-	            if(self.members.length == 6){
-                        async.series([
-                            playerRatingAsyncJob(self.members[0]),
-                            playerRatingAsyncJob(self.members[1]),
-                            playerRatingAsyncJob(self.members[2]),
-                            playerRatingAsyncJob(self.members[3]),
-                            playerRatingAsyncJob(self.members[4]),
-                            playerRatingAsyncJob(self.members[5]),
-                        ], function(err){
-                            if(!err){
-                                bot.reply(message, prepareMembersResponse(self));
-                            }else{
-                                bot.reply(message, err);
-                            }
-                        });
-                    }else{
-                        bot.reply(message, "I could not find that team.");
-                    }
-                });
+/* team members */
+chesster.hears({
+    middleware: [slack.requiresLeague],
+	patterns: ['team members'],
+    messageTypes: [
+        'direct_mention', 
+        'direct_message'
+    ]
+},
+function(bot, message) {
+    return Q.fcall(function() {
+        var teamName = message.text.split(" ").slice(2).join(" ");
+        if(teamName && teamName != ""){
+            message.league.formatTeamMembersResponse(teamName).then(function(response) {
+                bot.reply(message, response);
             });
         }else{
             bot.reply(message, "Which team did you say? [ team members <team-name> ]. Please try again.");
         }
     });
 });
 
-/* LICHESS STUFF */
-
-function getPlayerByName(name, callback){
-    var url = "http://en.lichess.org/api/user/" + name;
-    fetch_url_into_json(url, callback);
-}
-
-function getClassicalRating(opp, callback){
-  callback(opp.perfs.classical.rating);
-}
-
 /* welcome */
 
-controller.on('user_channel_join', function(bot, message) {
+chesster.on({event: 'user_channel_join'},
+function(bot, message) {
     bot_exception_handler(bot, message, function(){
         if(message.channel == channels.getId("general")){
             bot.reply(message, "Everyone, please welcome the newest member of the " 
                              + "Lichess 45+45 League, <@" + message.user + ">!");
             
             bot.startPrivateConversation(message, function(err, convo){
+                // TODO: the config links references below are hard coded
```

Comment:
Were you thinking all leagues could be displayed or perhaps a more interactive menu that would dive in to a particular league's links?